codegen/src/main/java/software/amazon/awssdk/codegen/IntermediateModelBuilder.java

```diff
@@ -107,12 +107,20 @@ public IntermediateModel build() {
             new HashMap<>(new AddCustomAuthorizers(this.service, getNamingStrategy()).constructAuthorizers());
 
         OperationModel endpointOperation = null;
+        boolean endpointCacheRequired = false;
 
         for (OperationModel o : operations.values()) {
             if (o.isEndpointOperation()) {
                 endpointOperation = o;
-                break;
             }
```

Comment:
We just use the last endpoint operation? Is that the right behavior?

Reply:
I'll admit this is not how I would have written it if I had done it myself, but it is actually a copy/paste from the v1 SDK. The logic is correct because it only needs to know there is at least one endpoint operation (and the last one suffices for that purpose), and the flag as to whether an endpoint is required is cumulative (once it gets set it doesn't get unset) so ends up effectively meaning 'at least one endpoint is required'.

Reply:
I added a couple of tests to assert this logic was working correctly as well.

Reply:
Can we clarify it a bit and make it a boolean or something?

Reply:
Changing it will require more refactoring than the change is worth in my opinion. At least it's in sync with v1 in this state. I'll add comments to explain what it's doing.